src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -852,6 +840,26 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
         return c;
     }
 
+    public static StandardCredentials resolveCredentials(Job job, String credentialsId, String url) {
```

Comment:
AFAIK that's only extracted the functionality into a method without a functional change - right?
I suggest to extract this change into a dedicated commit to separate functional changes from refactorings.